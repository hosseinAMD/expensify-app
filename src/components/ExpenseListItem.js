import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = (props) => (
    <div>
        <Link to={`/edit/${props.expense.id}`}><h3>{props.expense.description}</h3></Link>
        <p>{`${props.expense.amount} Rials`}
        --
        {moment(props.expense.createdAt).format('MMMM Do, YYYY')}</p>
    </div>
);

export default ExpenseListItem;