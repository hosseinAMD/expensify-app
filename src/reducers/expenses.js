//Expenses Reducer

const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
    if(action.type === 'ADD_EXPENSE'){
        return [...state,action.expense];
    } else if(action.type === 'REMOVE_EXPENSE'){
        return state.filter(({id}) => id !== action.id);
    } else if(action.type === 'EDIT_EXPENSE'){
        return state.map((expense) => {
            if(expense.id === action.id){
                return {
                    ...expense,
                    ...action.updates
                };
            } else {
                return expense;
            }
        });
    }
     else {
        return state;
    }
};

export default expensesReducer;