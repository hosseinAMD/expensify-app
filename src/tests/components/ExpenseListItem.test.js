import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('shou;d render individual expense itme',() => {
    const wrapper = shallow(<ExpenseListItem expense={expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
})