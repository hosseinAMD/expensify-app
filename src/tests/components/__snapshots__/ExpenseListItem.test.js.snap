// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shou;d render individual expense itme 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Book
    </h3>
  </Link>
  <p>
    25,000 Rials
    --
    January 1st, 1970
  </p>
</div>
`;
